pkg/utils/quantity.go

```diff
@@ -0,0 +1,90 @@
+package utils
+
+import (
+	"errors"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+var (
+	//It will map suffixes to make compatible with k8s "resource" package
+	transformSuffix = map[string]string{
+		"Ki": "Ki",
+		"K":  "Ki",
+		"k":  "Ki",
+		"Mi": "Mi",
+		"M":  "Mi",
+		"m":  "Mi",
+		"Gi": "Gi",
+		"G":  "Gi",
+		"g":  "Gi",
+	}
+)
+
+// parseQuantityString is a fast scanner for quantity values.
+// it extracts the numerical value and suffix from the quantity string
+func parseQuantityString(str string) (value, suffix string, err error) {
```

Comment:
Why would duplicate any of this code which looks to be copied from the kubernetes instead of simply using "MustParse" https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go#L125

Reply:
@jcantrill : Code in quantity.go doing `pre-processing` to convert suffix like `G->Gi` `M->Mi` and then using code from https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go#L264 to convert to bytes. So **1G** is converted to **1,073,741,824**.  If we directly invoke MustParse then 1G will be converted to 1,000,000,000.

Reply:
The best I can tell, the code introduced here almost exactly duplicates https://github.com/kubernetes/apimachinery/blob/master/pkg/api/resource/quantity.go#L125 which means we are supporting a copy that can vary and be subject to defects.  The code from APIMachinery has significantly larger visibility to a broader range of projects and is more likely to be correct or fixed if errors occur.   Please refactor the code to reuse the existing library

Reply:
This file is removed now.